### PR TITLE
Add tabs to profile (+ convert settings form submissions to AJAX)

### PR DESCRIPTION
**Pull request by THargreaves**
Add tabs to the profile page as per #36. In doing this, all form submissions must be switch to AJAX to avoid the tabset reseting to the first pane after each form submission (and corresponding redirect). This solves #24 in the process.

### CODE REVIEW

**Comment by THargreaves** [comment]
Just need to add AJAX requests for the instruments panel and then good to merge.

File: templates/profile/themes.html [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  <input type="hidden" name="csrf_token" value="{{ csrf_token() }}">
+  <input type="hidden" name="user_id" value="{{ current_user.id }}">
+  <div class="form-group">
+    <textarea class="form-control mb-4" id="theme_suggestions" name="theme_suggestions"
+      rows="3">{{ current_user.theme_suggestions }}</textarea>
+  </div>
+  <div class="d-grid gap-2">
+    <button type="submit" class="btn btn-primary btn-block mb-4" disabled id="theme_suggestions_button">Save</button>
+  </div>
+</form>
```
Comment (carlashworth) [inline templates/profile/themes.html:37]: add empty line?

File: templates/profile/instruments.html [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            {% endif %}
+        </div>
+
+    </div>
+    {% endfor %}
+</ul>
+{% else %}
+<p>It looks like you haven't auditioned for any instruments yet. You can book an audition slot from the Auditions tab.
+</p>
+{% endif %}
```
Comment (carlashworth) [inline templates/profile/instruments.html:60]: extra empty line?

File: templates/profile/notifications.html [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        data: form.serialize(),
+        success: function () {},
+        error: function (xhr, status, error) {
+          var err = eval("(" + xhr.responseText + ")");
+          alert(err.Message);
+        }
+      });
+    });
+  });
+</script>
```
Comment (carlashworth) [inline templates/profile/notifications.html:84]: "

File: templates/profile.html [hunk 4]
```diff
@@ -1,179 +1,36 @@
...
+    id="{{ tab_name }}" 
+    role="tabpanel"
+    aria-labelledby="{{ tab_name }}-tab">
+    {% include "profile/" + tab_name + ".html" %}
+  </div>
+  {% endfor %}
+</div>
+{% else %}
+Login
+{% endif %}
```
Comment (carlashworth) [inline templates/profile.html:35]: line at end?

File: templates/profile/themes.html [hunk 5]
```diff
@@ -0,0 +1,37 @@
+<h3 class="mb-2 mt-5">Theme Suggestions</h3>
+<p>Suggest themes and setlist ideas that you would like to see us play. If we select a theme that is similar to your
+  suggestion, you will be more likely to be selected for the band</p>
```
Comment (carlashworth) [inline templates/profile/themes.html:3]: potentially same line

**Review by carlashworth — CHANGES_REQUESTED** [review]
Perfect functionally, what a banger of a PR. Just (I might be wrong) a few spacing changes, but then just merge that, really nice changes.

File: templates/profile/themes.html (hunk 5 above)
Comment (THargreaves) [reply templates/profile/themes.html:3]: Personally, I'm a fan of following a sensible recommended line length (~120 max):
https://stackoverflow.com/questions/2886603/is-there-a-recommended-maximum-line-length-for-html-or-javascript

**Pull request merged**